__tests__/validations/component.test.tsx

```diff
@@ -115,13 +115,165 @@ test('Throw when one of the prefabs options is invalid', (t: Context): void => {
+              type: 'PARTIAL',
+              descendants: [],
+            },
+          ],
+        },
+      ],
+    },
+  ] as unknown as Prefab[];
+
+  t.throws(() => validatePrefabs(prefabs));
+});
+test('Throw when type partial has name', (t: Context): void => {
+  const prefabs = [
+    {
+      name: 'Component Name',
+      icon: 'TitleIcon',
+      category: 'CONTENT',
+      structure: [
+        {
+          name: 'something',
+          options: [
+            {
+              value: '',
+              label: 'something',
+              key: 'something',
+              type: 'TEXT',
+            },
+          ],
+          descendants: [
+            {
+              name: 'Partial',
+              type: 'PARTIAL',
+            },
+          ],
+        },
+      ],
+    },
+  ] as unknown as Prefab[];
+
+  t.throws(() => validatePrefabs(prefabs));
+});
+test('Throw when type partial has options', (t: Context): void => {
+  const prefabs = [
+    {
+      name: 'Component Name',
+      icon: 'TitleIcon',
+      category: 'CONTENT',
+      structure: [
+        {
+          name: 'something',
+          options: [
+            {
+              value: '',
+              label: 'something',
+              key: 'something',
+              type: 'TEXT',
+            },
+          ],
+          descendants: [
+            {
+              type: 'PARTIAL',
+              options: [],
+            },
+          ],
+        },
+      ],
+    },
+  ] as unknown as Prefab[];
+
+  t.throws(() => validatePrefabs(prefabs));
+});
+test('Throw when type component has id', (t: Context): void => {
+  const prefabs = [
+    {
+      name: 'Component Name',
+      icon: 'TitleIcon',
+      category: 'CONTENT',
+      structure: [
+        {
+          name: 'something',
+          options: [
+            {
+              value: '',
+              label: 'something',
+              key: 'something',
+              type: 'TEXT',
+            },
+          ],
+          descendants: [
+            {
+              name: 'Component',
+              partialId: '',
+              options: [
+                {
+                  type: 'PARTIAL_REFERENCE',
+                  label: 'Partial Reference',
+                  key: 'partialReferenceId',
+                  value: '""',
+                },
+                {
+                  type: 'PARTIAL_INPUT_OBJECTS',
+                  label: 'Partial inputs',
+                  key: 'partialInputMapping',
+                  value: '{}',
+                },
+              ],
+              descendants: [],
+            },
+          ],
+        },
+      ],
+    },
+  ] as unknown as Prefab[];
 
   t.throws(() => validatePrefabs(prefabs));
 });
+test('Does not throw when partial object is within the structure', (t: Context): void => {
+  const prefabs = [
+    {
+      name: 'Component Name',
+      icon: 'TitleIcon',
+      category: 'CONTENT',
+      structure: [
+        {
+          type: 'PARTIAL',
```

Comment:
When a reference of type `PARTIAL` does not have an `id` key with a value the validation should probably throw?

Reply:
I made the partialId required but it still can be empty